src/ReactiveSocket.cpp

```diff
@@ -35,14 +35,15 @@ ReactiveSocket::ReactiveSocket(
     bool isServer,
     std::shared_ptr<RequestHandlerBase> handler,
     Stats& stats,
-    std::unique_ptr<KeepaliveTimer> keepaliveTimer)
+    std::unique_ptr<KeepaliveTimer> keepaliveTimer,
+    folly::Executor& executor)
     : handler_(handler),
       connection_(std::make_shared<ConnectionAutomaton>(
-          [handler](
+          [this, handler](
```

Comment:
Theoretically it is possible that ConnectionAutomaton will outlive ReactiveSocket, but even in that case ConnectionAutomaton will never call back into this method. 
Either way I am changing this in the next diff, because I also dont like "this" to be captured here. So the problem will go away and in the meantime it will not cause any harm.